pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -402,6 +407,60 @@ public BrokerService(PulsarService pulsar, EventLoopGroup eventLoopGroup) throws
         this.bundlesQuotas = new BundlesQuotas(pulsar.getLocalMetadataStore());
     }
 
+    public void addTopicEventListener(TopicEventsListener... listeners) {
+        Objects.requireNonNull(listeners);
+        Arrays.stream(listeners)
+                .filter(x -> x != null)
+                .forEach(topicEventListeners::add);
+        getTopics().keys().forEach(topic ->
+                notifyTopicListeners(listeners, topic, TopicEvent.LOAD, EventStage.SUCCESS, null));
+    }
+
+    public void removeTopicEventListener(TopicEventsListener... listeners) {
+        Objects.requireNonNull(listeners);
+        Arrays.stream(listeners)
+                .filter(x -> x != null)
+                .forEach(topicEventListeners::remove);
+    }
+
+    private void notifyTopicListeners(String topic, TopicEvent event, EventStage stage) {
+        notifyTopicListeners(topic, event, stage, null);
+    }
+
+    private void notifyTopicListeners(String topic, TopicEvent event, EventStage stage, Throwable t) {
+        for (TopicEventsListener listener : topicEventListeners) {
+            try {
+                listener.handleEvent(topic, event, stage, t);
+            } catch (Throwable ex) {
+                log.error("TopicEventsListener exception while handling {}_{} for topic {}", event, stage, topic, ex);
```

Comment:
nit:
in case of InterruptedException we should call Thread.currentThread().interrupt();

Reply:
This is not applicable here. handleEvent does not define checked exceptions (InterruptedException is a checked exception) so the event listener will have to deal with that. This exception handling is for cases like RuntimeException/NPE/..